crates/breez-sdk/core/src/sdk.rs

```diff
@@ -1510,6 +1677,205 @@ impl BreezSdk {
         Box::pin(self.send_payment_internal(request, false)).await
     }
 
+    pub async fn fetch_convert_token_limits(
+        &self,
+        request: FetchConvertTokenLimitsRequest,
+    ) -> Result<FetchConvertTokenLimitsResponse, SdkError> {
+        self.ensure_spark_private_mode_initialized().await?;
+        let (asset_in_address, asset_out_address) = match request.convert_type {
+            ConvertType::FromBitcoin {
+                to_token_identifier,
+            } => (BTC_ASSET_ADDRESS.to_string(), to_token_identifier),
+            ConvertType::ToBitcoin {
+                from_token_identifier,
+            } => (from_token_identifier, BTC_ASSET_ADDRESS.to_string()),
+        };
+        let min_amounts = self
+            .flashnet_client
+            .get_min_amounts(GetMinAmountsRequest {
+                asset_in_address,
+                asset_out_address,
+            })
+            .await?;
+        Ok(FetchConvertTokenLimitsResponse {
+            min_from_amount: min_amounts.asset_in_min,
+            min_to_amount: min_amounts.asset_out_min,
+        })
+    }
+
+    /// Prepares a conversion by validating the request and estimating the receive amount and fee.
+    ///
+    /// # Arguments
+    ///
+    /// * `request` - The prepare convert token request containing the amount, token identifier, and convert type.
+    ///
+    /// # Returns
+    ///
+    /// Result containing either:
+    /// * `PrepareConvertTokenResponse` - The response containing the estimated receive amount and fee.
+    /// * `SdkError` - If there was an error during the preparation process.
+    pub async fn prepare_convert_token(
+        &self,
+        request: PrepareConvertTokenRequest,
+    ) -> Result<PrepareConvertTokenResponse, SdkError> {
+        self.ensure_spark_private_mode_initialized().await?;
+        let (asset_in_address, asset_out_address) = self
+            .validate_convert_token_params(request.amount, &request.convert_type)
+            .await?;
+        let pools_response = self
+            .flashnet_client
+            .list_pools(ListPoolsRequest {
+                asset_a_address: Some(asset_in_address.clone()),
+                asset_b_address: Some(asset_out_address.clone()),
+                sort: Some(PoolSortOrder::Volume24hDesc),
+                ..Default::default()
+            })
+            .await?;
+        // Get the pool for the conversion
+        let pool = pools_response.pools.first().ok_or(SdkError::Generic(
+            "No pool found for the given token identifier".to_string(),
+        ))?;
+        let response = self
+            .flashnet_client
+            .simulate_swap(SimulateSwapRequest {
+                asset_in_address: asset_in_address.clone(),
+                asset_out_address: asset_out_address.clone(),
+                pool_id: pool.lp_public_key,
+                amount_in: request.amount,
+                integrator_bps: None,
+            })
+            .await?;
```

Comment:
Now that we don't need to sync we can treat flashnet as a swapper (swap provider) from the sdk perspective and this code (and some more )could be extracted out to its own swapper trait/service. It it a nice to have for now that will get the flashnet specifics out of the sdk generic code. I wouldn't hold the merge for this feedback.

Reply:
I agree but I would rather wait for another implementation before attempting to generalize as a plugin. This has complexities also around integration with the payment list, sync etc... perhaps it is too soon.